system/jlib/jsocket.cpp

```diff
@@ -4860,6 +4866,10 @@ class CSocketEpollThread: public CSocketBaseThread
         }
         CriticalBlock block(sect);
         removeSock(sock);
+        unsigned n = items.ordinality();
+        // new handler thread
+        if (n >= MAX_EPOLL_FD_EVENTS)
```

Comment:
is the meaning/use of MAX_EPOLL_FD_EVENTS being overloaded here?
It's used to epoll_create - the manual says it's advisory and since 2.6.8 it's ignored but must be >1
It's used to epoll_wait - to cap the number of events returned in one go, which may be reasonable.

But the code here is capping to the number added, which needn't be the same as the max number of events epoll_wait returns.

I think should differentiate between max addable to handle and max events returned in one shot.

Reply:
also, is a 5k limit sensible here when it was 32k in pre-epoll days?

Reply:
I did overload it - I have separated them now.  The number of events returned probably would not be more than the number of fds (one event per fd ...) so I initially thought reasonable to use same macro value.
I wanted to spread out the notifySelected handling into more threads than currently, so I went with 5k instead of 32k.  But this PR is also related to an HPCC 2018 Offsite topic discussion and how to refactor/design for optimal notifySelected msg loop event handling (espeically with SSL in mind)